src/mpl/include/mpl_gpu.h

```diff
@@ -0,0 +1,39 @@
+/*
+ *  Copyright (C) by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#ifndef MPL_GPU_H_INCLUDED
+#define MPL_GPU_H_INCLUDED
+
+#include "mplconfig.h"
+
+#ifdef MPL_HAVE_GPU
+
+#ifdef MPL_HAVE_CUDA
+#include "mpl_gpu_cuda.h"
+#elif defined MPL_HAVE_HIP
+#include "mpl_gpu_hip.h"
+#else
+#error "No GPU runtime library available"
+#endif
+
+#define MPL_GPU_SUCCESS 0
+#define MPL_GPU_FAILURE 1
```

Comment:
Why do we have GPU-specific SUCCESS/FAILURE, instead of generic `MPL_SUCCESS`?  We should unify this all through MPL.

Reply:
We don't have an `MPL_SUCCESS` yet. The current convention with MPL is for each components to define its own. Maybe we should unify that?

Reply:
Right.  That's my suggestion too.

Reply:
Added one commit to unify all the MPL return codes.